### PR TITLE
HW-3 | Implement Sorted Storage. Test it. Refactor error logs

### PR DESCRIPTION
**Pull request by laroself1**

### CODE REVIEW

File: src/storage/ArrayStorage.java [hunk 1]
```diff
@@ -6,72 +6,103 @@
...
 
+    @Override
     public void save(Resume r) {
         Objects.requireNonNull(r);
         if (size == MAX_STORAGE_SIZE) {
-            log.warning("Storage max limit is reached. Please, clear storage to add new elements.");
+            logStorageMaxLimit();
             return;
         }
         if (isPresent(r.getUuid())) {
```
Comment (den-ko) [inline src/storage/ArrayStorage.java]: There is no need for `getResumeIndex()` method invocation twice. It's called in the `isPresent()` and in the code below

File: src/storage/ArrayStorage.java [hunk 2]
```diff
@@ -6,72 +6,103 @@
...
+        int index = getResumeIndex(uuid);
         if (isValidIndex(index)) {
             return storage[index];
         }
-        log.warning(String.format("Resume(uuid=%s) is not present in storage. Return null.", uuid));
+        logNotFound(uuid);
         return null;
     }
 
+    protected void logEmptyUUID() {
```
Comment (den-ko) [inline src/storage/ArrayStorage.java]: This method can be deleted. Use `logNotFound()`

**Review by den-ko — CHANGES_REQUESTED** [review]

File: src/storage/ArrayStorage.java (hunk 1 above)
Comment (laroself1) [reply src/storage/ArrayStorage.java]: Done

File: src/storage/ArrayStorage.java (hunk 2 above)
Comment (laroself1) [reply src/storage/ArrayStorage.java]: Done

**Pull request merged**